pkg/process/runner/exclusive.go

```diff
@@ -0,0 +1,88 @@
+package runner
+
+import (
+	"context"
+	"os"
+	"sync"
+
+	"github.com/leptonai/gpud/pkg/log"
+	"github.com/leptonai/gpud/pkg/process"
+)
+
+var _ Runner = &exclusiveRunner{}
+
+func NewExclusiveRunner() Runner {
+	return &exclusiveRunner{}
+}
+
+// exclusiveRunner is a scheduler that runs a single process at a time.
+// Does not support concurrent execution of multiple processes.
+type exclusiveRunner struct {
+	mu      sync.RWMutex
+	running process.Process
+}
+
+// RunUntilCompletion starts a bash script, blocks until it finishes,
+// and returns the output and the exit code.
+// If there is already a process running, it returns an error.
+func (er *exclusiveRunner) RunUntilCompletion(ctx context.Context, script string) ([]byte, int32, error) {
+	if er.alreadyRunning() {
+		return nil, 0, ErrProcessAlreadyRunning
+	}
+
+	// write all stderr + stdout to a temporary file
+	tmpFile, err := os.CreateTemp(os.TempDir(), "gpud-process-output-*.txt")
```

Comment:
I forgot that if there is an issue whether write the output to a bytes buffer..

Reply:
You mean, piping the output to memory and return? I think we can... but like you said, we may also hit the buffer limit, whereas writing to a file doesn't have that kind of issue

Reply:
When we want script output stream, maybe we can read line by line and send them back, but if we assume, we always run until completion, I think reading from the file is simpler. let me know if you have other preferences.

Reply:
OK. Let's output to a file. By the way, we should consider that if the GPU crashes, there may be no chance to delete the file and we should find a way to clean it.

Reply:
@eahydra done.